src/transpose.js

```diff
@@ -0,0 +1,41 @@
+var _curry1 = require('./internal/_curry1');
+/**
+ * Transposes the rows and columns of a 2D list.
+ * When passed a list of `n` lists of length `x`,
+ * returns a list of `x` lists of length `n`.
+ *
+ *
+ * @func
+ * @memberOf R
+ * @since v0.19.0
+ * @category List
+ * @sig transpose :: [[a]] -> [[a]]
+ * @param {Array} list A 2D list
+ * @return {Array} A 2D list
+ * @example
+ *
+ *      R.transpose([[1, 'a'], [2, 'b'], [3, 'c']]) //=> [[1, 2, 3], ['a', 'b', 'c']]
+ *      R.transpose([[1, 2, 3], ['a', 'b', 'c']]) //=> [[1, 'a'], [2, 'b'], [3, 'c']]
+ *
+ * If some of the rows are shorter than the following rows, their elements are skipped:
+ *
+ *      R.transpose([[10, 11], [20], [], [30, 31, 32]]) //=> [[10, 20, 30], [11, 31], [32]]
+ */
+module.exports = _curry1(function transpose(outerlist) {
+  var outlen = outerlist.length;
+  var x = 0;
+  var result = [];
+  while (x < outlen) {
+    var innerlist = outerlist[x];
+    var y = 0;
+    var inlen = innerlist.length;
+    while (y < inlen) {
+      result[y] = result[y] || [];
```

Comment:
Could you replace the `||` with an explicit check? Using `||` with non-Boolean operands is a common source of bugs (and has been responsible for half a dozen Ramda bugs).